File: src/main/java/dev/interfiber/karpet/server/entitys/MinecraftEntity.java
package dev.interfiber.karpet.server.entitys;

import net.minestom.server.attribute.Attribute;
import net.minestom.server.entity.EntityCreature;
import net.minestom.server.entity.EntityType;

/**
 *
 * @author persephone
 */

public class MinecraftEntity extends EntityCreature {
    public MinecraftEntity(EntityType entityType, EntityStats stats){
        super(entityType);
        getAttribute(Attribute.MAX_HEALTH).setBaseValue(stats.MaxHealth);
        getAttribute(Attribute.ATTACK_DAMAGE).setBaseValue(stats.AttackDamage);
    }
}

File: src/main/kotlin/dev/interfiber/karpet/server/entitys/MinecraftEntity.kt
package dev.interfiber.karpet.server.entitys

import net.minestom.server.attribute.Attribute
import net.minestom.server.entity.EntityCreature
import net.minestom.server.entity.EntityType

/**
 * Represents a entity in minecraft
 * @author Interfiber
 */
open class MinecraftEntity(entityType: EntityType?, stats: EntityStats) : EntityCreature(entityType!!) {
    init {
        getAttribute(Attribute.MAX_HEALTH).baseValue = stats.maxHealth.toFloat()
        getAttribute(Attribute.ATTACK_DAMAGE).baseValue = stats.attackDamage.toFloat()
    }
}